### PR TITLE
KIALI-1403 Add error handling to the side panels when loading sparklines

### PR DESCRIPTION
**Pull request by israel-hdez**
Now, a warning is shown in the side panel if data cannot be loaded.

peek 2018-08-28 17-05

### CODE REVIEW

**Comment by israel-hdez** [comment]
The side panels may need some refactor. Very similar patterns are on all panels.

File: src/pages/ServiceGraph/SummaryPanelEdge.tsx [hunk 1]
```diff
@@ -217,14 +232,19 @@ export default class SummaryPanelEdge extends React.Component<SummaryPanelPropTy
         });
       })
       .catch(error => {
-        if (!this._isMounted) {
-          console.log('SummaryPanelEdge: Ignore fetch error, component not mounted.');
+        if (error.isCancelled) {
```
Comment (jshaughn) [inline src/pages/ServiceGraph/SummaryPanelEdge.tsx]: I think this is actually wrong, as you are using 'isCanceled' in the utils.

Comment (israel-hdez) [reply src/pages/ServiceGraph/SummaryPanelEdge.tsx]: Ops, yes. Good catch.
All references changed.

File: src/pages/ServiceGraph/ServiceGraphPage.tsx [hunk 2]
```diff
@@ -14,6 +14,7 @@ import GraphFilterToolbar from '../../components/GraphFilter/GraphFilterToolbar'
 import { computePrometheusQueryInterval } from '../../services/Prometheus';
 import { style } from 'typestyle';
 
+import { CancellablePromise, makeCancelablePromise } from '../../utils/Common';
```
[the remark this answers is not in the document]
Comment (israel-hdez) [reply src/pages/ServiceGraph/ServiceGraphPage.tsx]: Fixed. Using only one 'i'.
Usually I use two 'l's. I didn't know about the new preferred wording. I'm falling behind :(

[the remark this answers is not in the document]
Comment (jshaughn) [reply src/pages/ServiceGraph/ServiceGraphPage.tsx]: Well maybe it's only preferred by me :-).

**Comment by jshaughn** [comment]
@israel-hdez I approve the PR but I am still wondering about why we don't use the message center.

**Comment by israel-hdez** [comment]
@jshaughn In #616 now the message is shown in the page. In #616 the same error message is also pushed to the message center. With that PR, I thought that it was going to be preferred to show the error in-place; I'm omitting the message center, since I think it's redundant.
I was just following the last work done. But we should talk a little about it and consolidate behaviors (I hope this to be a quick decision). I don't have preference of one over the other. Consistency is what I consider important.

**Comment by jshaughn** [comment]
Perhaps inquire with UX,  Merging for now...

**Pull request merged**